libraries/src/CMS/Application/ApiApplication.php

```diff
@@ -134,5 +133,38 @@ public function getTemplate($params = false)
 		// The API application should not need to use a template
 		return 'system';
 	}
-}
 
+	/**
+	 * Route the application.
+	 *
+	 * Routing is the process of examining the request environment to determine which
+	 * component should receive the request. The component optional parameters
+	 * are then set in the request object to be processed when the application is being
+	 * dispatched.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function route()
+	{
+		$uri = \JUri::getInstance();
+		$router = static::getRouter();
+
+		// Trigger the onBeforeApiRoute event.
+		PluginHelper::importPlugin('webservices');
+		$this->triggerEvent('onBeforeApiRoute', &$router);
+	}
+
+	/**
+	 * Returns the application Router object.
+	 *
+	 * @return  ApiRouter
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public static function getApiRouter()
```

Comment:
We can make this method non-static, then you can change the call in the method to `$this->getContainer()->get('ApiRouter');`